sdk/eventgrid/azure-eventgrid/azure/eventgrid/_operations/_patch.py

```diff
@@ -187,6 +254,140 @@ def receive_cloud_events(
         receive_result_deserialized = ReceiveResult(value=detail_items)
         return receive_result_deserialized
 
+    def _publish_binary_mode(self, topic_name: str, event: Any, api_version, **kwargs: Any) -> None:
+
+        error_map = {
+            401: ClientAuthenticationError, 404: ResourceNotFoundError, 409: ResourceExistsError, 304: ResourceNotModifiedError
+        }
+        error_map.update(kwargs.pop('error_map', {}) or {})
+
+        _headers = case_insensitive_dict(kwargs.pop("headers", {}) or {})
+        _params = kwargs.pop("params", {}) or {}
+
+        cls: ClsType[_models._models.PublishResult] = kwargs.pop(  # pylint: disable=protected-access
+            'cls', None
+        )
+
+        content_type: str = kwargs.pop('content_type', _headers.pop('content-type', "application/cloudevents+json; charset=utf-8"))
+
+        # Given that we know the cloud event is binary mode, we can convert it to a HTTP request
+        http_request = _to_http_request(            
+            topic_name=topic_name,
+            api_version=api_version,
+            headers=_headers,
+            params=_params,
+            content_type=content_type,
```

Comment:
are there any content types that wouldn't work with binary_mode? and if so, would the service return an error if you pass in binary data and invalid content type?

Reply:
yes it would if the datacontenttype is incorrect you would get a 'cannot deserialize input value' from service